### ISSUE TITLE
Kraken Crash

### ISSUE THREAD

**Issue by MiningInMySleep** [ctubio/Krypto-trading-bot#622]
```
- currency: ETH/JPY
- roll-out: 1517755208230
- lastbeat: 1517755229137
- os-uname: Linux 4.13.0-0.bpo.1-amd64 #1 SMP Debian 4.13.13-1~bpo9+1 (2017-11-22) x86_64
- tracelog:
./app/server/K(_ZN1K2EV3wtfEi+0x301) [0x5651accc0711]
/lib/x86_64-linux-gnu/libc.so.6(+0x33030) [0x7f210f6f0030]
./app/server/K(_ZN1K2SH6logErrENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES6_S6_+0x18) [0x5651accbe0a8]
./app/server/K(_ZN1K2SH6logWarENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES6_+0x7a) [0x5651accbe37a]
./app/server/K(_ZN1K2FN4wJetENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEbS6_S6_S6_+0x2bb) [0x5651acd5727b]
./app/server/K(_ZN1K8GwKraken11sync_ordersEv+0x867) [0x5651acd59f27]
./app/server/K(_ZNSt17_Function_handlerIFSt10unique_ptrINSt13__future_base12_Result_baseENS2_8_DeleterEEvENS1_12_Task_setterIS0_INS1_7_ResultISt6vectorIN1K6mOrderESaISA_EEEES3_ESt12_Bind_simpleIFZNKS9_2Gw6ordersMUlvE_clEvEUlvE_vEESC_EEE9_M_invokeERKSt9_Any_data+0x25) [0x5651acd40245]
./app/server/K(_ZNSt13__future_base13_State_baseV29_M_do_setEPSt8functionIFSt10unique_ptrINS_12_Result_baseENS3_8_DeleterEEvEEPb+0x1b) [0x5651accb69ab]
/lib/x86_64-linux-gnu/libpthread.so.0(+0xe739) [0x7f210fa6a739]
./app/server/K(_ZNSt6thread11_State_implISt12_Bind_simpleIFZNSt13__future_base17_Async_state_implIS1_IFZNK1K2Gw6ordersMUlvE_clEvEUlvE_vEESt6vectorINS4_6mOrderESaISB_EEEC4EOS9_EUlvE_vEEE6_M_runEv+0xd2) [0x5651acd3c8a2]
./app/server/K(+0x54347f) [0x5651ad08047f]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x7494) [0x7f210fa63494]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f210f7a5aff]
```
Also running into the issue where 2/5 of the pairs I'm trading have this error:
Warrrrning: Unable to calculate TBP, missing wallet data.

I've tried doing manual buys/sells, but the issue persists.

Edit: Actually, what's even stranger is after doing a make restartall, the 2 pairs that were originally working fine are having the same issue.

**Comment by ctubio** [comment]
and "make reinstall" doen't help?

try running only 1 bot in Kraken, and see if running more bots generates this issue
thanks'¡

**Comment by MiningInMySleep** [comment]
This was after a fresh make reinstall after I saw that the "Kraken Support" issue was closed.
Running with just ETH/USD alone still has the same issue.
Tried doing another make reinstall just to be sure, but still nothing.

**Comment by ctubio** [comment]
thanks for the info, lemme try on ETH/USD; i can't reproduce this issue yet :S

**Comment by ctubio** [comment]
maybe if you run with --naked there is some interesting info before the error about curl failing?

**Comment by ctubio** [comment]
in last commit i had updated CA bundle file of curl, just to discard the problem is not with the certs

**Comment by MiningInMySleep** [comment]
Ok... So I left it running overnight and at some point in the last 6 hours it started working. Probably exchange lagging or having issues?

**Comment by ctubio** [comment]
i will try to avoid segfaults when curl response code is diff than 200 soOn

**Comment by MiningInMySleep** [comment]
I think it might be some kind of rate limitation on total number of API calls.
I'm sure you've already read the official limitations on their API help page here: https://www.kraken.com/help/api
My guess is that once you hit this limit, you get a longer timeout period which refreshes each time you attempt an API call during said timeout period. The reason I think this is the case is that I had one instance that repeatedly received this error over and over for about 8 hours straight. On a whim, I tried stopping the bot for about a minute and then restarted it and everything worked fine.
I'm running some tests on my end to see how many trading pairs I can run without issue, but I would imagine that you'd have better tools to diagnose this than I would.

**Comment by ctubio** [comment]
this looks like #615 and #625 , seems a curl error not properly displayed and instead crashing, will try to fix handling of curl errors soOn, so is easily readable what is going on (lets fix it at #625)
